ui/lib/CreateStream.tsx

```diff
@@ -75,58 +111,120 @@ const CreateStream = (props: CreateStreamProps) => {
+    setIsApproved(true);
+  };
+
+  const renderBalanceCheck = () => {
+    // Step 1: Check gas and native token balance
+    if (gasPrice === null || balance === null) {
+      return (
+        <div id="CreateStream">
+          <p>(Step 1/2) Checking gas and native token balance...</p>
+        </div>
+      );
+    }
+
+    // Step 2: Check if there are enough native tokens for fees
+    if (gasPrice >= balance) {
+      return (
+        <div id="CreateStream">
+          <p> Not enough native tokens to pay for transaction fees</p>
+          <button
+            onClick={(e) => {
+              window.location.href = "https://hop.exchange/";
+            }}
+          >
+            Go to Hop Exchange
+          </button>
+        </div>
+      );
+    }
+
+    // Step 3: Check stream token balance
+    if (transactionAmount === null || tokenBalance === null) {
+      return (
+        <div id="CreateStream">
+          <p>(Step 2/2) Checking stream token balance...</p>
+        </div>
+      );
+    }
+
+    // Step 4: Check if the user has enough stream tokens
+    if (transactionAmount >= tokenBalance) {
+      return (
+        <div id="CreateStream">
+          <p> Not enough tokens to pay for stream</p>
+          <button
+            onClick={(e) => {
+              window.location.href = "https://app.uniswap.org/";
+            }}
+          >
+            Go to Uniswap
+          </button>
+        </div>
+      );
+    }
+
+    // Step 5: If all checks pass, show the modal
+    return (
+      <div id="CreateStream">
+        <div>
+          {/* Display: What is the product they are paying for? With what token? */}
+          {/* @ts-ignore */}
+          {props.cart && props.cart}
+        </div>
+        {/* Part 1: The user needs to decide how long they want the stream to run for  */}
+        <div id="CreateStream-lifetime">
+          <Slider
+            min={1}
+            marks={marks}
+            max={maxTimeDays}
+            step={null}
+            value={Math.min(selectedTime / SECS_PER_DAY, maxTimeDays)}
+            defaultValue={1}
+            onChange={(value) =>
+              typeof value === "number" && setSelectedTime(SECS_PER_DAY * value)
+            }
+          />
+        </div>
+        <br />
+        {/* Part 2: The user approves exactly X tokens, corresponding to stream life   */}
+        <div id="CreateStream-approve">
+          <button disabled={!approveStream} onClick={onApproveClick}>
+            {`Approve ${transactionAmount} ${tokenData?.symbol}`}
+          </button>
+        </div>
+        <br />
+        {/* Part 3: The user creates their stream, using the given reason/product code */}
+        <div id="CreateStream-create">
+          <button disabled={!isApproved} onClick={createStream}>
+            {`Open Stream for ${selectedTime / SECS_PER_DAY} day${
+              selectedTime !== SECS_PER_DAY ? "s" : ""
+            }`}
+          </button>
+        </div>
```

Comment:
Wonder if we want to make this a multi-part sequence:
1. Confirm cart and select payment token (TODO item, just USDC for now... maybe make it a dropdown that just displays USDC)
2. ("Confirm" in CreateStream =>) Select desired stream length based on selected token and stream cost
3. ("Approve" in CreateStream => wait for approval transaction =>) Review stream parameters and execute stream creation (which updates via `props.registerStream`)